### PR TITLE
Unique constraint now takes in table name in postgres and mssql

### PR DESCRIPTION
**Pull request by andrew-w-ross**
Added mssql to docker-compose
typeorm/typeorm/issues#108

### CODE REVIEW

File: gulpfile.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
-import {Gulpclass, Task, SequenceTask, MergedTask} from "gulpclass";
+import { Gulpclass, Task, SequenceTask, MergedTask } from "gulpclass";
+
+declare function require(module: string): any;
```
Comment (pleerock) [inline gulpfile.ts]: don't think you should do it. `require` should be defined and comes from `@types/node` package. Why did you add it?

File: docker-compose.yml [hunk 2]
```diff
@@ -34,4 +34,16 @@ services:
     environment:
       POSTGRES_USER: "test"
       POSTGRES_PASSWORD: "test"
-      POSTGRES_DB: "test"
\ No newline at end of file
+      POSTGRES_DB: "test"
+
+  # mssql
+  mssql:
```
Comment (pleerock) [inline docker-compose.yml]: nice, does it work under macos / linux?

File: test/github-issues/108/issue-108.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { expect } from "chai";
+
+describe("github issues > #108 Error with constraint names on postgres", () => {
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        schemaCreate: true,
+        dropSchemaOnConnection: true,
+        enabledDrivers: ["postgres", "mssql"] // TODO:Add tests for mssql
```
Comment (pleerock) [inline test/github-issues/108/issue-108.ts]: why did you enable only specific drivers, and not all of them?

File: test/github-issues/108/issue-108.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }));
+    after(() => closeTestingConnections(connections));
+
+    it("should sync and create column names in the form uk_{tableName}_{columnName}", () => Promise.all(connections.map(async connection => {
+        await connection.syncSchema(true);
```
Comment (pleerock) [inline test/github-issues/108/issue-108.ts]: because you have `schemaCreate: true, dropSchemaOnConnection: true,` connection settings sync schema already happens without this line of code

File: test/github-issues/108/issue-108.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        for (const entityType of [User, Project]) {
+            const metaData = connection.getMetadata(entityType);
+            const [nameCol] = metaData.columns.filter(cm => cm.name === "name");
+            expect(nameCol).not.undefined;
+            expect(nameCol.isUnique).to.be.true;
+        }
```
Comment (pleerock) [inline test/github-issues/108/issue-108.ts]: not sure if its a correct test... You probably shouldn't check it here because this test looks like it tests if metadatas are properly created from decorators

File: docker-compose.yml (hunk 2 above)
Comment (andrew-w-ross) [inline docker-compose.yml]: There are 2 images one for Linux and windows. I am using the Linux image one on my windows machine and it's working fine. Seems to break on osx though even though it should technically work.

File: test/github-issues/108/issue-108.ts (hunk 4 above)
Comment (andrew-w-ross) [inline test/github-issues/108/issue-108.ts]: Sure I'll remove it.

File: test/github-issues/108/issue-108.ts (hunk 5 above)
Comment (andrew-w-ross) [inline test/github-issues/108/issue-108.ts]: I had to change how a column is determined as unique so I put it in here. On second thought you're right the other tests should catch if I broke it.

File: docker-compose.yml (hunk 2 above)
Comment (pleerock) [inline docker-compose.yml]: would be great if it work on mac too....

File: gulpfile.ts (hunk 1 above)
Comment (pleerock) [inline gulpfile.ts]: hmm it should work without problems without this line. Maybe you have something outdated (including globally installed ts-node)? Or maybe old version of typescript, or your ide is not set to use latest version of compiler? Because it does compile without any errors on my local and CI too. 

```typescript
$ gulp tests
[09:21:18] Requiring external module ts-node/register
[09:21:21] Using gulpfile ~/www/opensource/typeorm/typeorm/gulpfile.ts
[09:21:21] Starting 'tests'...
[09:21:21] Starting 'compile'...
[09:21:28] Finished 'compile' after 7.22 s
[09:21:28] Starting 'tslint'...
[09:21:32] Finished 'tslint' after 3.8 s
[09:21:32] Starting 'coveragePre'...
[09:21:33] Finished 'coveragePre' after 908 ms
[09:21:33] Starting 'coveragePost'...
...
[09:22:32] Finished 'coveragePost' after 59 s
[09:22:32] Starting 'coverageRemap'...
[09:22:32] Finished 'coverageRemap' after 674 ms
[09:22:32] Finished 'tests' after 1.18 min
```

Everything else in this PR looks good

**Comment by pleerock** [comment]
thank you for this amazing PR 👍

**Comment by andrew-w-ross** [comment]
I definitely wouldn't call it amazing. But you're welcome.
